daemon/cluster/cluster.go

```diff
@@ -871,6 +872,36 @@ func (c *Cluster) GetServices(options apitypes.ServiceListOptions) ([]types.Serv
 	return services, nil
 }
 
+// imageWithDigestString takes an image such as name or name:tag
+// and returns the image pinned to a digest, such as name@sha256:34234...
+func (c *Cluster) imageWithDigestString(ctx context.Context, image, encodedAuth string) (string, error) {
```

Comment:
`encodedAuth` should really be cross-cutting. Perhaps, store in the backed or preconfigure an object with the auth before calling this method.